tonic/src/status.rs

```diff
@@ -45,6 +44,32 @@ pub struct Status {
     /// If the metadata contains any headers with names reserved either by the gRPC spec
     /// or by `Status` fields above, they will be ignored.
     metadata: MetadataMap,
+    /// Optional underlying `h2::Error` if this `Status` was converted from a `h2::Error`.
+    #[cfg(feature = "transport")]
+    h2_error: Option<h2::Error>,
```

Comment:
I would like to not make this the specific h2 error but actually a more general `Box<dyn Error + Send + sync>` and to expose that via the `Error::source` methods. Then a user can pull in `h2` and extract that error and do what they need with it.

Reply:
Yeah, I could see it down the line storing other errors which is why I wanted to go more generic.

The clone does put a wrench into this doesn't it...I think we could drop clone to simplify this. What do you think? We are already preparing a breaking release so we could include this. cc @davidpdrsn